managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -3445,6 +3445,32 @@ public PositionImpl getNextValidPosition(final PositionImpl position) {
         return next;
     }
 
+    public PositionImpl getValidPositionAfterSkippedEntries(final PositionImpl position, int skippedEntryNum) {
+        PositionImpl skippedPosition;
+        try {
+            skippedPosition = getValidPositionInternal(position, skippedEntryNum);
+        } catch (NullPointerException e) {
+            skippedPosition = lastConfirmedEntry.getNext();
+            if (log.isDebugEnabled()) {
+                log.debug("[{}] Can't find valid position : {}, fall back to the next position of the last "
+                        + "position : {}.", position, name, skippedPosition, e);
+            }
+        }
+        return skippedPosition;
+    }
+
+    public PositionImpl getValidPositionInternal(final PositionImpl position, int skippedEntryNum) {
+        PositionImpl toPosition = position.getPositionAfterEntries(skippedEntryNum);
+        while (!isValidPosition(toPosition)) {
+            Long nextLedgerId = ledgers.ceilingKey(toPosition.getLedgerId() + 1);
+            if (nextLedgerId == null) {
+                throw new NullPointerException();
```

Comment:
Throwing NPE is bad practice.
Please throw a new checked exception.
This way we have full control over what we are catching

Reply:
OK, I will fix

Reply:
Fixed!  PTAL,thanks! @eolivelli